sc-24sp/assignments/de02E/game-of-life/src/main.rs

```diff
@@ -0,0 +1,130 @@
+                if (*status == state) {
+                    neighbor_count += 1;
+                }
+            }
+        neighbor_count
+        }
+    }
+
+fn main() {
+    let uniHeight = 35;
+    let uniWidth = 35;
+    let mut cell_vec: Vec<CellState> = Vec::new();
+    for i in 0..uniHeight*uniWidth {
+        let rng = rand::thread_rng().gen_range(0..3);
+        match rng {
+            0 => {
+                cell_vec.push(CellState::Dead);
+            }
+            1 => {
+                cell_vec.push(CellState::Alive);
+            }
+            2 => {
+                cell_vec.push(CellState::Warm);
+            }
+            _ => {
+                print!("Error!");
+            }
+        }
+    }
+    let mut universe = Universe{width: uniWidth,height: uniHeight,cells: cell_vec.clone()};
+    loop {
+        let mut properIndex = (0,0);
+        for i in 0..uniHeight {
+            for j in 0..uniWidth {
+                match universe.get_status(universe.get_index(i, j)) {
+                    CellState::Alive => {
+                        print!("███");
+                    }
+                    CellState::Dead => {
+                        print!("░░░");
+                    }
+                    CellState::Warm => {
+                        print!("▒▒▒");
+                    }
+                }
+            }
+            println!();
+        }
+        for i in 0..uniHeight {
+            for j in 0..uniWidth {
+                let live_count = universe.live_neighbor_count(i, j, CellState::Alive);
+                let warm_count = universe.live_neighbor_count(i, j, CellState::Warm);
+                let cState = &mut cell_vec[universe.get_index(i, j)];
+                // Rule 1: Overcrowding
+                if (live_count < 2) && *cState == CellState::Alive {*cState = CellState::Warm;}
+                // Rule 2: Survival
+                else if (live_count < 4 && live_count >= 2) && *cState == CellState::Alive {*cState = CellState::Alive;}
+                // Rule 3: Overcrowding
+                else if (live_count >= 4) && *cState == CellState::Alive {*cState = CellState::Warm;}
+                // Rule 4: Reproduction
+                else if (live_count == 3) && *cState == CellState::Dead {*cState = CellState::Alive;}
+                // Rule 5: Ressurection
+                else if (live_count == 2) && *cState == CellState::Warm {*cState = CellState::Alive;}
+                // Rule 6: Decay
+                else if warm_count < 6 && *cState == CellState::Warm {*cState = CellState::Dead}
+                // Rule 7: Preservation
+                else if (warm_count < 8 && live_count >= 6) && *cState == CellState::Warm {*cState = CellState::Dead}
+                // Rule 8: Abiogenesis
+                else if warm_count == 8 && *cState == CellState::Warm {*cState = CellState::Alive}
+            }
+        }
+        
+        universe = Universe{width: uniHeight,height: uniWidth,cells: cell_vec.clone()};
```

Comment:
why transpose `width` and `height` here with `uniWidth` and `uniHeight`?

`cell_vec` is already a new empty Vec, why not use it directly instead of calling `.clone()`?